### PR TITLE
DGS-9904: Update SR Describe CLI v4 to call srcm/v3 API

### PR DESCRIPTION
**Pull request by GunalKupta**
Release Notes
-------------
No breaking changes

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

DO NOT MERGE until after https://github.com/confluentinc/ccloud-sdk-go-v2/pull/142

What
----
Migrate the SR cluster describe command to call srcm/v3 because srcm/v2 will soon be deprecated as part of [Always-On Governance](https://confluentinc.atlassian.net/wiki/spaces/DG/pages/3120630044/1-Pager+Always+On+Stream+Governance+Phase+2#Clusters-(srcm%2Fv2))
Also upgraded org/v2 SDK 

References
----------
[DGS-9904](https://confluentinc.atlassian.net/browse/DGS-9904)

Test & Review
-------------

Ran:
- `make build`
- `make unit-test`
- `make lint`
- `dist/confluent_$(go env GOOS)_$(go env GOARCH)/confluent login --url https://stag.cpdev.cloud/`
- `dist/confluent_$(go env GOOS)_$(go env GOARCH)/confluent sr cluster describe`

[DGS-9904]: https://confluentinc.atlassian.net/browse/DGS-9904?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/ccloudv2/srcm.go
Comment (GunalKupta) [inline pkg/ccloudv2/srcm.go:151]: An alternative implementation is to just replace this srcm/v2 API call with srcm/v3 instead of creating a separate function to call srcm/v3, but that would affect a bunch of other components. Is it worth the risk?

**Comment by brianstrauch** [comment]
> No breaking changes

Want to merge this into main instead of v4, then?

File: go.mod
Comment (brianstrauch) [inline go.mod:14]: Make sure you use confluentinc/ccloud-sdk-go-v2 (public)!

File: pkg/ccloudv2/srcm.go
Comment (brianstrauch) [inline pkg/ccloudv2/srcm.go:13]: Why are we still using V2? Hasn't everything switched over to V3?

File: pkg/ccloudv2/srcm.go
Comment (brianstrauch) [reply pkg/ccloudv2/srcm.go:151]: > but that would affect a bunch of other components

What do you mean by this?

File: go.mod
Comment (GunalKupta) [reply go.mod:14]: srcm/v3 is not in the public repo yet... looks like this PR will have to wait 😅

File: pkg/ccloudv2/srcm.go
Comment (GunalKupta) [reply pkg/ccloudv2/srcm.go:13]: Not yet, srcm/v2 has not been deprecated yet

**Comment by GunalKupta** [comment]
> What do you mean by this

The GetSchemaRegistryClustersByEnvironment function is called by 7 different packages, so migrating the SRCM API could have unintended consequences

File: pkg/ccloudv2/srcm.go
Comment (brianstrauch) [reply pkg/ccloudv2/srcm.go:13]: Got it, hopefully the plan is to include all of the V2 endpoints used in the CLI in V3!

**Comment by brianstrauch** [comment]
> > What do you mean by this
> 
> The GetSchemaRegistryClustersByEnvironment function is called by 7 different packages, so migrating the SRCM API could have unintended consequences

We should do it as long as there are no breaking changes or bugs in the v3 implementation :)

File: pkg/ccloudv2/srcm.go
Comment (GunalKupta) [reply pkg/ccloudv2/srcm.go]: just realized that srcm/v2 will be deprecated when srcm/v3 is released (by definition) 🤦‍♂️

File: go.mod
Comment (brianstrauch) [inline go.mod:36]: Can we add a new version tag instead of using a commit?

Comment (GunalKupta) [reply go.mod:36]: This is temporary until the PR is merged: https://github.com/confluentinc/ccloud-sdk-go-v2/pull/142

File: pkg/ccloudv2/client.go
Comment (brianstrauch) [inline pkg/ccloudv2/client.go:54]:
```suggestion
	SrcmClient                  *srcmv3.APIClient
```

File: test/test-server/srcm_handlers.go
Comment (brianstrauch) [inline test/test-server/srcm_handlers.go:12]: Can we get rid of this dependency now?

**Review by brianstrauch — COMMENTED** [review]
LGTM pending https://github.com/confluentinc/cli/pull/2585/files#r1475082394

**Review by brianstrauch — APPROVED** [review]

**Review by sgagniere — APPROVED** [review]
+1

**Review by akhileshm1 — APPROVED** [review]

**Pull request merged**
